src_c/surflock.c

```diff
@@ -115,36 +115,52 @@ pgSurface_UnlockBy(pgSurfaceObject *surfobj, PyObject *lockobj)
     pgSurfaceObject *surf = (pgSurfaceObject *)surfobj;
     int found = 0;
     int noerror = 1;
+    int weakref_getref_result;
 
     if (surf->locklist != NULL) {
         PyObject *item, *ref;
         Py_ssize_t len = PyList_Size(surf->locklist);
         while (--len >= 0 && !found) {
             item = PyList_GetItem(surf->locklist, len);
-            ref = PyWeakref_GetObject(item);
-            if (ref == lockobj) {
-                if (PySequence_DelItem(surf->locklist, len) == -1) {
-                    return 0;
-                }
-                else {
-                    found = 1;
+
+            weakref_getref_result = PyWeakref_GetRef(item, &ref);
+            if (weakref_getref_result == -1) {
+                noerror = 0;
+            }
+            if (weakref_getref_result == 1) {
+                if (ref == lockobj) {
+                    if (PySequence_DelItem(surf->locklist, len) == -1) {
+                        Py_DECREF(ref);
+                        return 0;
+                    }
+                    else {
+                        found = 1;
+                    }
                 }
+                Py_DECREF(ref);
             }
         }
 
         /* Clear dead references */
         len = PyList_Size(surf->locklist);
         while (--len >= 0) {
             item = PyList_GetItem(surf->locklist, len);
-            ref = PyWeakref_GetObject(item);
-            if (ref == Py_None) {
+
+            weakref_getref_result = PyWeakref_GetRef(item, &ref);
+            if (weakref_getref_result == -1) {
+                noerror = 0;
+            }
+            if (ref == NULL) {
```

Comment:
```suggestion
            if (weakref_getref_result == 0) {
```
`ref == NULL` would catch the error case too, right?

Reply:
Well the docs don't say it will write NULL on error, that detail is undefined. But it could grab a null from a previous iteration of the loop, assuming nothing is written. Adjusted the logic.